### ISSUE TITLE
PS1 (both lists) error with "require a login or archive returned no file" even when login is set correctly

### ISSUE THREAD

**Issue by archon810** [zach-morris/plugin.program.iagl#37]
Is this a known issue?

**Comment by archon810** [comment]
Same with Sega Saturn.

Dreamcast, on the other hand, says no emulators found, even though Retroarch has it installed (Reicast).

**Comment by archon810** [comment]
This is on Android TV btw.

**Comment by zach-morris** [comment]
It works for me.  I just checked both.  You likely have bad credentials entered.  Keep in mind you need to enter a valid **email** and the password.

For dreamcast, it sounds like you didn't setup launching for Retroarch/external launching and it's trying to play the game with Retroplayer.  Check the [Wiki](https://github.com/zach-morris/plugin.program.iagl/wiki/1.--Setup).

**Comment by zach-morris** [comment]
1.  Select the game list, bring up the context menu and choose 'View List Settings'
Under launch parameters, it should say:  Launch With:  External
And then there should be a command listed that looks like the following:
```
/system/bin/am start --user 0 -a android.intent.action.MAIN -c android.intent.category.LAUNCHER -e ROM "%ROM_PATH%" -e LIBRETRO /data/data/com.retroarch/cores/reicast_libretro_android.so -e CONFIGFILE %CFG_PATH% -e IME com.android.inputmethod.latin/.LatinIME -n com.retroarch/.browser.retroactivity.RetroActivityFuture
```

2.  I'm not sure about the credentials thing.  Like I said, I tried those two game lists and they work.  I went a step further and removed my credentials, and then I get the error your getting.  You can provide a log, but I'm guessing it says your credentials are not correct, or archive.org returned a file that was not valid.  You can go one step further and try and manually download the file in a browser.  If you look in your kodi debug log, it will state something like this:
```
NOTICE: IAGL:  Attempting to download file https://archive.org/download/blahblahblah.zip
```

**Comment by archon810** [comment]
Deleted the previous comment. There was a typo in the email when I tried a PC for a test after all. The PS1 lists are working here. I'll go back and check the Android TV setup now and try to get logs from there.

I'll get back to you on the Dreamcast launcher issue.

**Comment by zach-morris** [comment]
Closing since the original issue is resolved

**Comment by archon810** [comment]
Log from Android TV (Shield TV). I've triple checked the credentials - same ones that work on Windows:

```
12:13:43.712 T:149840053488  NOTICE: IAGL:  Attempting to download file https://archive.org/download/RedumpSonyPlayStationAmerica20160617/Castlevania%20-%20Symphony%20of%20the%20Night%20%28USA%29.zip as logged in user
12:13:43.712 T:149840053488  NOTICE: IAGL:  Saving file to /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/userdata/addon_data/plugin.program.iagl/temp_iagl/Castlevania - Symphony of the Night (USA).zip as logged in user
12:13:43.713 T:149840053488   DEBUG: DialogProgress::Open called
12:13:43.713 T:149840053488   DEBUG: ------ Window Init (DialogConfirm.xml) ------
12:13:43.713 T:149840053488    INFO: Loading skin file: DialogConfirm.xml, load type: KEEP_IN_MEMORY
12:13:44.355 T:149840053488  NOTICE: IAGL:  File saved to location /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/userdata/addon_data/plugin.program.iagl/temp_iagl/Castlevania - Symphony of the Night (USA).zip, file size 4211, file crc 6E0490CF
12:13:44.364 T:149840053488   DEBUG: IAGL:  Checking downloaded file /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/userdata/addon_data/plugin.program.iagl/temp_iagl/Castlevania - Symphony of the Night (USA).zip validity
12:13:44.367 T:149840053488   DEBUG: IAGL:  The archive /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/userdata/addon_data/plugin.program.iagl/temp_iagl/Castlevania - Symphony of the Night (USA).zip returned a bad file, deleted.  This archive is either no longer available or requires login credentials to be entered into IAGL settings.
```

**Comment by archon810** [comment]
Jesus, I figured it out finally. Somehow the email had 2 invisible spaces added before the email address. They're completely non obvious because of the way Kodi UI works (it saves previous input, doesn't show spaces in a good way, Android TV is not input-friendly). Interesting that the same issue happened on both of my Shield TVs. Perhaps there's a space in the email field by default or something.

**Comment by archon810** [comment]
Just spent a couple hours on the Dreamcast issue.

1. For some reason, the plugin set it to Kodi's player, even though I rescanned downloaded Retroarch cores and made sure the Reicast core is in Retroarch.

2. I switched it manually to external Retroarch and tried all 3 Retroarch start settings. Retroarch crashed on start every time. 

3. Even though it's not officially available for Android TV on the Play Store, I sideloaded Reicast stable (beta would crash), set it up (it needs additional setup for dc_bios and dc_flash), paired the controllers, and then pointed it at the Kodi game cache folder. This worked finally.

4. I tried to update the plugin's launch command to reicast direct (without retroarch), but that fails / crashes like before. Not sure why.

Any ideas?
